tests/APITests/APITest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Test\APITests;
+
+use App\Database\ConnectionManager;
+use App\Database\QueryBuilder;
+use App\Database\QueryWriterFactory;
+use App\Mapper\WordMapper;
+use App\Model\Word;
+use App\Repository\WordRepository;
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\TestCase;
+
+class APITest extends TestCase
+{
+    private static \PDO $connection;
+
+    public static function setUpBeforeClass(): void
+    {
+        self::$connection = ConnectionManager::getConnection();
+    }
+
+    public function tearDown(): void
+    {
+        $query = 'SET FOREIGN_KEY_CHECKS=0;'
+            . 'TRUNCATE TABLE words;'
+            . 'SET FOREIGN_KEY_CHECKS=1;';
+        self::$connection->exec($query);
+    }
+
+    public static function wordProvider(): array
+    {
+        return [
+            [
+                [
+                    new Word('mistranslate'),
+                    new Word('products'),
+                    new Word('university'),
+                    new Word('management'),
+                ]
+            ],
+            [
+                [
+                    new Word('technology'),
+                ]
+            ],
+            [
+                []
+            ]
+        ];
+    }
+
+    #[DataProvider('wordProvider')]
+    public function testListWordsApi(array $words): void
+    {
+        $factory = new QueryWriterFactory();
+        $builder = new QueryBuilder($factory);
+        $wordRepository = new WordRepository(self::$connection, $builder);
+        $wordRepository->insertWords($words);
+
+        $inserted = $wordRepository->getWords();
+        $mapper = new WordMapper();
+        $expected = array_map(fn (Word $item): array => $mapper->serialize($item), $inserted);
+
+        $url = 'http://localhost:8000/api/words';
```

Comment:
http://localhost:8000 should be configurable and not hard coded. 

It could be taken from env, or maybe just injected somewhere or at very least it could be a constant.

Why? because URL could change, lets say you have to change port or `localhost` becomes reserved, now searching everywhere it is being used will be pain